### PR TITLE
Correct description of OTEL_EXPORTER_JAEGER_ENDPOINT and OTEL_TRACES_EXPORTER

### LINKED ISSUE #2331: Clarify how the default value of OTEL_EXPORTER_JAEGER_ENDPOINT should work
**What are you trying to achieve?**
My organization is trying to switch from opentracing to opentelemetry, and currently the .net exporter will not work with our infrastructure, because our jaeger collector listens on a different path than `/api/traces`. With the [opentracing jaeger client](https://github.com/jaegertracing/jaeger-client-csharp), we were able to specify the endpoint and have everything wired up correctly. With the current spec, the default value of the `OTEL_EXPORTER_JAEGER_ENDPOINT` env var makes it seem like libraries need to append `/api/traces` themselves to correctly send spans over http. 

What did you expect to see?
I'd like it to be clear in the spec how this should behave - should libraries append `/api/traces` - or should the default value include the path in the uri, and expect developers to configure the full url to export to. If libraries *should* append the path, can we add another configuration option to override it (this is what my organization needs to adopt otel in .net).

**Additional context.**
The .net sdk is currently implemented to append `/api/traces` to the endpoint, but the golang, javascript, and ruby sdks all default the value of the endpoint to include the path, and require users to specify the path in the value.
I opened a PR to try to make the .net sdk configurable: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2847
This led to a discussion where one of the maintainers felt that the golang implementation is wrong: https://github.com/open-telemetry/opentelemetry-go/issues/2599

### PR DESCRIPTION
**Pull request by yurishkuro**
Fixes #2331

## Changes

The OTEL_EXPORTER_JAEGER_ENDPOINT env var description is updated:
  * clarify that it refers to Thrift over HTTP endpoint, provide a link to Jaeger API spec
  * change the default value to point to the correct HTTP port, not the gRPC port
  * clarify that the path (`/api/traces`) must be included in the value, to support installations running at remapped URLs

The OTEL_TRACES_EXPORTER env var description updated to reflect state of the world where all but Java SDKs implement http/thrift protocol, not gRPC as was stated in the description.

### CODE REVIEW

**Comment by carlosalberto** [comment]
@yurishkuro Should we get this as part of 1.9, or should we hold this? (Sounds like the former?)

**Comment by yurishkuro** [comment]
@pellared I do not believe that `OTEL_TRACES_EXPORTER=jaeger` == gRPC is actually respected by the SDKs, because Java is the only SDK that even has a `+` against Protobuf/gRPC in the compliance matrix, whereas almost all SDKs support Thrift/HTTP. So I would suggest changing the description for `OTEL_TRACES_EXPORTER=jaeger` to say something like "usually HTTP, sometimes gRPC, please check the SDK docs". Then introducing OTEL_EXPORTER_JAEGER_PROTOCOL should be a separate PR since (a) I don't know the level of consensus around it, and (b) this PR is mostly about correcting the wrong value for HTTP endpoint that all (except Java) SDKs implement currently.

**Comment by pellared** [comment]
> @pellared I do not believe that `OTEL_TRACES_EXPORTER=jaeger` == gRPC is actually respected by the SDKs, because Java is the only SDK that even has a `+` against Protobuf/gRPC in the compliance matrix, whereas almost all SDKs support Thrift/HTTP. So I would suggest changing the description for `OTEL_TRACES_EXPORTER=jaeger` to say something like "usually HTTP, sometimes gRPC, please check the SDK docs". Then introducing OTEL_EXPORTER_JAEGER_PROTOCOL should be a separate PR since (a) I don't know the level of consensus around it, and (b) this PR is mostly about correcting the wrong value for HTTP endpoint that all (except Java) SDKs implement currently.

It could be done similarly to the OTLP Exporter. Something like:

```
[*] If no configuration is provided the default protocol SHOULD be `http/thrift` unless SDKs have good reasons to choose `grpc` as the default (e.g. for backward compatibility reasons when `grpc` was already the default in a stable SDK release).
```

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -128,7 +128,7 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
 |---------------------------------|------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
 | OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent                                    | "localhost"                                                                                      |
 | OTEL_EXPORTER_JAEGER_AGENT_PORT | Port for the Jaeger agent `compact` Thrift protocol              | 6831                                                                                             |
-| OTEL_EXPORTER_JAEGER_ENDPOINT   | HTTP endpoint for Jaeger traces                                  | <!-- markdown-link-check-disable --> "http://localhost:14250"<!-- markdown-link-check-enable --> |
+| OTEL_EXPORTER_JAEGER_ENDPOINT   | Full URL of [Thrift over HTTP][[jaeger_http]] endpoint for Jaeger traces | <!-- markdown-link-check-disable --> "http://localhost:14268/api/traces"<!-- markdown-link-check-enable --> |
```
Comment (pellared) [inline specification/sdk-environment-variables.md:131]: What if gRPC protocol is used? Can this env var be used?

File: specification/sdk-environment-variables.md [hunk 2]
```diff
@@ -179,7 +179,11 @@ The SDK MAY accept a comma-separated list to enable setting multiple exporters.
 Known values for `OTEL_TRACES_EXPORTER` are:
 
 - `"otlp"`: [OTLP](./protocol/otlp.md)
-- `"jaeger"`: [Jaeger gRPC](https://www.jaegertracing.io/docs/1.21/apis/#protobuf-via-grpc-stable)
+- `"jaeger"`: export in Jaeger data model. If no additional configuration is
+  provided the default protocol SHOULD be [Thrift over HTTP][jaeger_http] unless
+  SDKs have good reasons to choose [Protobuf/gRPC][jaeger_grpc] as the default
```
Comment (pellared) [inline specification/sdk-environment-variables.md:184]:
```suggestion
  SDKs have good reasons to choose [gRPC][jaeger_grpc] as the default
```

File: specification/sdk-environment-variables.md [hunk 3]
```diff
@@ -126,14 +126,16 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
-| OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent                                    | "localhost"                                                                                      |
+| OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent (1)                                | "localhost"                                                                                      |
 | OTEL_EXPORTER_JAEGER_AGENT_PORT | Port for the Jaeger agent `compact` Thrift protocol              | 6831                                                                                             |
-| OTEL_EXPORTER_JAEGER_ENDPOINT   | HTTP endpoint for Jaeger traces                                  | <!-- markdown-link-check-disable --> "http://localhost:14250"<!-- markdown-link-check-enable --> |
+| OTEL_EXPORTER_JAEGER_ENDPOINT   | Full URL of [Thrift over HTTP][[jaeger_http]] endpoint for Jaeger traces (2) | <!-- markdown-link-check-disable --> "http://localhost:14268/api/traces"<!-- markdown-link-check-enable --> |
```
Comment (pellared) [inline specification/sdk-environment-variables.md:131]: something is wrong with the hyperlink formatting here

File: specification/sdk-environment-variables.md [hunk 4]
```diff
@@ -126,14 +126,16 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
-| OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent                                    | "localhost"                                                                                      |
+| OTEL_EXPORTER_JAEGER_AGENT_HOST | Hostname for the Jaeger agent [1]                                | "localhost"                                                                                      |
 | OTEL_EXPORTER_JAEGER_AGENT_PORT | Port for the Jaeger agent `compact` Thrift protocol              | 6831                                                                                             |
-| OTEL_EXPORTER_JAEGER_ENDPOINT   | HTTP endpoint for Jaeger traces                                  | <!-- markdown-link-check-disable --> "http://localhost:14250"<!-- markdown-link-check-enable --> |
+| OTEL_EXPORTER_JAEGER_ENDPOINT   | Full URL of [Thrift over HTTP][jaeger_http] endpoint for Jaeger traces [2] | <!-- markdown-link-check-disable --> `"http://localhost:14268/api/traces"` <!-- markdown-link-check-enable --> |
```
Comment (pellared) [inline specification/sdk-environment-variables.md:131]: image

File: specification/sdk-environment-variables.md [hunk 5]
```diff
@@ -126,14 +126,16 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
 | OTEL_EXPORTER_JAEGER_TIMEOUT    | Maximum time the Jaeger exporter will wait for each batch export | 10s                                                                                              |
 | OTEL_EXPORTER_JAEGER_USER       | Username to be used for HTTP basic authentication                |                                                                                                  |
 | OTEL_EXPORTER_JAEGER_PASSWORD   | Password to be used for HTTP basic authentication                |                                                                                                  |
 
-See [Jaeger Agent](https://www.jaegertracing.io/docs/latest/deployment/#agent) documentation.
+[1] See [Jaeger Agent](https://www.jaegertracing.io/docs/latest/deployment/#agent) documentation.
+
+[2] If SDK defaults to using gRPC, then OTEL_EXPORTER_JAEGER_ENDPOINT refers to gRPC endpoint, e.g. `http://localhost:14250`.
```
Comment (pellared) [inline specification/sdk-environment-variables.md:138]: How about something more or less like

```suggestion
[2] When the exporter uses the gRPC protocol, then it refers to the gRPC endpoint. In such a case, the default value should be `http://localhost:14250`.
```

**Comment by carlosalberto** [comment]
@pellared Does the latest commit fix your issue?

**Review by pellared — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Pull request merged**